extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/query/SparqlFluoQueryBuilder.java

```diff
@@ -417,6 +438,60 @@ public void meet(final Projection node) {
             // Walk to the next node.
             super.meet(node);
         }
+        
+        
+        public void meet(Reduced node) {
+            //create id, initialize ConstructQueryMetadata builder, register ConstructQueryMetadata 
+            //builder with FluoQueryBuilder, and add metadata that we currently have
+            final String constructId = nodeIds.getOrMakeId(node);
+            final ConstructQueryMetadata.Builder constructBuilder = ConstructQueryMetadata.builder();
+            constructBuilder.setNodeId(constructId);
+            fluoQueryBuilder.setConstructQueryMetadata(constructBuilder);
+            constructBuilder.setSparql(sparql);
+            
+            //get child node
+            QueryModelNode child = node.getArg();
+            Preconditions.checkArgument(child instanceof Projection || child instanceof MultiProjection);
```

Comment:
Think I just forgot that I had statically imported checkNotNull.  Didn't look to see if I was consistent.